pkg/metrics/gauge.go

```diff
@@ -0,0 +1,53 @@
+package metrics
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Gauge is a metric type that represents a single numerical value that can
+// arbitrarily go up and down.
+type Gauge struct {
+	name   string
+	labels map[string]string
+
+	value     float64
+	timestamp int64
+	mutex     sync.RWMutex
+}
+
+// Set allows setting the gauge to an arbitrary value.
+func (g *Gauge) Set(value float64) {
+	g.mutex.Lock()
+	defer g.mutex.Unlock()
+
+	g.value = value
+	g.timestamp = time.Now().UnixNano() / int64(time.Millisecond)
```

Comment:
I think it's better (safer) if we calculate it as:
```
time.Nanosecond * time.Now().UnixNano() / time.Millisecond
```

Reply:
Can you specify why exactly?

Reply:
It works because `time` package works with nanosecond precision (`Nanosecond  Duration = 1`), so doing `int64(time.Millisecond)` gives `1e6` but if that ever changes, we'll not be converting properly here. See https://stackoverflow.com/a/34860368/577812

Reply:
Ok thanks! But your proposition needs to covert `time.Now().UnixNano()` to `time.Duration` as well. So, maybe we should just use: `time.Duration(time.Now().UnixNano()) / time.Millisecond`?

Reply:
Or maybe just the simplest possible way: `g.timestamp = time.Now().UnixNano() / 1e6` - we know that we need to divide by 1e+6 when going from millisecond to nanosecond, it's independent of how `time` package precision.

Let's also maybe add a comment to `timestamp int64` field that it's in milliseconds.

Reply:
Sure!